src/tickit/core/management/schedulers/base.py

```diff
@@ -54,24 +65,29 @@ async def update_component(self, input: Input) -> None:
         """
         await self.state_producer.produce(input_topic(input.target), input)
 
-    async def handle_message(self, message: Union[Interrupt, Output]) -> None:
-        """A callback to handle interrupts or outputs produced by the state consumer.
+    async def handle_message(
+        self, message: Union[Interrupt, Output, ComponentException]
+    ) -> None:
+        """A callback to handle interrupts, outputs or exceptions produced by the state
+        consumer.
 
-        An asynchronous callback which handles interrupt and output messages produced by
-        the state consumer; For Outputs, changes are propagated and wakeups scheduled
-        if required, whilst handling of interrupts is deferred.
+        An asynchronous callback which handles interrupt, Output and ComponentException
```

Comment:
`Interrupt` should also be made proper here

```suggestion
        An asynchronous callback which handles Interrupt, Output and ComponentException
```